cocos/3d/framework/mesh-renderer.ts

```diff
@@ -374,6 +374,31 @@ export class MeshRenderer extends ModelRenderer {
         this._updateCastShadow();
     }
 
+    /**
+     * @en Is received direction Light.
+     * @zh 是否接收平行光光照。
+     * @param visibility @en direction light visibility. @zh 方向光的可见性。
+     */
+    public onUpdateReceiveDirLight (visibility: number) {
+        if (!this._model) { return; }
+        if (this.node && ((visibility & this.node.layer) === this.node.layer)
+        || (visibility & this._model.visFlags)) {
+            if (this._shadowCastingMode === ModelShadowCastingMode.OFF) {
+                this._model.castShadow = false;
+            } else {
+                assertIsTrue(
+                    this._shadowCastingMode === ModelShadowCastingMode.ON,
+                    `ShadowCastingMode ${this._shadowCastingMode} is not supported.`,
+                );
+                this._model.castShadow = true;
+            }
+            this._model.receiveDirLight = true;
```

Comment:
add light bake settings:
1. if(MainLight.mobility==static && (model._isBaked==true || sceneGlobal.lightProbeInfo.data.hasCoefficients && model._useLightProbe=true)) this._model.receiveDirLight=false;
2. "Finished lightmap bakeing, Change dirlight mobility, Change use light probe" need invoke onUpdateReceiveDirLight function
3. Only 1 DirLight Can be add to scene